### PR TITLE
connect to API

### PR DESCRIPTION
**Pull request by Somdotta07**
- Connected with  Bookstore API
- Refactored the codes
- Used middleware

### CODE REVIEW

File: src/components/BooksPage.js [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+        {bookList.map((book) => <Book key={book.item_id} book={book} />)}
       </ul>
       <form id="add-book-form" onSubmit={(e) => submitForm(e)}>
-        <input id="title" placeholder="Book Title" />
-        <input id="author" placeholder="Book Author" />
+        <input id="title" placeholder="Name" />
+        <input id="category" placeholder="Category" />
         <button type="submit"> Add Book </button>
       </form>
     </div>
```
Comment (TedLivist) [inline src/components/BooksPage.js]: Please take a look at the image below
- [x] I have deleted one book but it is still on the redux store and on the page. Your changes are not stored correctly on the redux store when I add or delete a book. I need to reload the browser for the store to be updated from the server.
  
  Remember, this project is merely a refactor of the previous one and so just like your previous `create-remove` project, the books list should be updated without page reload. Kindly fix this behavior.

  somdo-books

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Somdotta07,

Good job so far!
✔️ Good documentation
✔️ Nicely modularized code
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Somdotta07** [comment]
Fixed the problem observed by the previous reviewer.

**Review by dasileker — APPROVED** [review]
## **Hi @Somdotta07 ,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**